src/plot-data.ts

```diff
@@ -2045,21 +2056,29 @@ export abstract class PlotData {
     }
     this.tooltip_list = [];
     if (this.type_ == 'scatterplot') {
-      for (let i=0; i<this.plotObject.point_list.length; i++) {
+      for (let i = 0; i < this.plotObject.point_list.length; i++) {
         this.plotObject.point_list[i].selected = false;
         if (this.scatter_points[i]) {
           this.scatter_points[i].selected = false;
           if (reset_clicked_points) this.scatter_points[i].clicked = false;
         }
       }
     } else if (this.type_ == 'graph2d') {
-      for (let i=0; i<this.plotObject.graphs.length; i++) {
-        for (let j=0; j<this.plotObject.graphs[i].point_list.length; j++) {
+      for (let i = 0; i < this.plotObject.graphs.length; i++) {
+        for (let j = 0; j < this.plotObject.graphs[i].point_list.length; j++) {
           this.plotObject.graphs[i].point_list[j].selected = false;
           this.plotObject.graphs[i].point_list[j].clicked = false;
         }
       }
-    }
+    } else if (this.plotObject instanceof PieChart) {
```

Comment:
Exact same comment as the previous one here, but with a function.

Globally speaking, wherever there are a lot of "if/else if/else" statements, there is almost everytime a better way to do it by using Oriented Object stuff.

See dc's schema refactor for an example